proxy/upstreams.go

```diff
@@ -25,7 +25,7 @@ type UpstreamConfig struct {
 // So the following config: ["[/host.com/]1.2.3.4", "[/www.host.com/]2.3.4.5", "[/maps.host.com/]#", "3.4.5.6"]
 // will send queries for *.host.com to 1.2.3.4, except for *.www.host.com, which will go to 2.3.4.5 and *.maps.host.com,
 // which will go to default server 3.4.5.6 with all other domains
-func ParseUpstreamsConfig(upstreamConfig, bootstrapDNS []string, timeout time.Duration) (UpstreamConfig, error) {
+func ParseUpstreamsConfig(upstreamConfig, bootstrapDNS []string, timeout time.Duration, options upstream.Options) (UpstreamConfig, error) {
```

Comment:
Both bootstrapDNS and timeout can be also passed inside `upstream.Options`

Reply:
Added in 783ae89